src/test/java/de/tum/in/www1/artemis/metis/linkpreview/LinkPreviewIntegrationTest.java

```diff
@@ -66,6 +81,22 @@ void testLinkPreviewDataExtraction(String url) throws Exception {
         }
     }
 
+    /**
+     * Mocks the OpenGraph object and its content for the given url.
+     *
+     * @param url the url
+     */
+    private void mockOpenGraph(String url) {
+        Mockito.when(ogParser.getOpenGraphOf(url)).thenReturn(mockOpenGraph);
```

Comment:
I don't know what these ogParser methods do, but usually doReturn is the better option for Spies as with when the methods are still executed.
https://site.mockito.org/javadoc/current/org/mockito/Spy.html